libdirections-session/src/main/java/com/mapbox/navigation/directions/session/MapboxDirectionsSession.kt

```diff
@@ -54,12 +68,18 @@ class MapboxDirectionsSession(private val router: Router) : DirectionsSession {
     }
 
     private fun requestRoute() {
-        if (::origin.isInitialized && ::waypoints.isInitialized) {
-            router.cancel()
-            currentRoute = null
-            router.getRoute(origin, waypoints) {
-                currentRoute = it
+        router.cancel()
+        router.getRoute(origin, waypoints, destination, object : Router.Callback {
+            override fun onRouteReady(routes: List<Route>) {
+                val route = routes.firstOrNull()
+                currentRoute = route
+                routeObservers.forEach { it.onRouteChanged(route) }
             }
-        }
+
+            override fun onFailure(throwable: Throwable) {
+                currentRoute = null
```

Comment:
If we are assigning `null` here, we have to also call `routeObservers.forEach { it.onRouteChanged(null) }`.

Reply:
hm, I'm not sure, we send `onFailure` event, so it looks enough for current state of `route fetching`

Reply:
But we change the state of the variable, therefore, we need to notify the observers.

Reply:
yes, you're right. And we notify observer by another callback `onFailure`.  I think that we need another opinion 
@Guardiola31337 @Lebedinsky wdyt?